src/popup/popup.js

```diff
@@ -1 +1,64 @@
-/* popup.js */
+async function getAllExtensions() {
+  const extensions = await browser.management.getAll();
+  return extensions.filter((extension) => extension.type !== 'theme');
+}
+
+async function areExtsBeingMonitored() {
+  const { status } = await browser.runtime.sendMessage({
+    getMonitoringStatus: true,
+  });
+  return status || false;
+}
+
+async function initializeMonitorAll(extensions) {
+  await browser.runtime.sendMessage({
+    extStartMonitorAll: extensions,
+  });
+  window.close();
+}
+
+async function stopMonitorAll() {
+  await browser.runtime.sendMessage({
+    extStopMonitorAll: true,
+  });
+  window.close();
+}
+
+async function viewActivityLogs() {
+  browser.tabs.create({
+    url: `${browser.runtime.getURL('../activitylog/activitylog.html')}`,
+  });
+  window.close();
+}
+
```

Comment:
Move methods that don't interact with the view to a different class/file. This makes a clearer module structure and allows you to reuse that code.